metadata-load/src/main/java/uk/ac/ebi/ampt2d/metadata/importer/ObjectsImporter.java

```diff
@@ -142,6 +162,55 @@ public Analysis importAnalysis(String accession) {
     protected abstract Analysis extractStudyFromAnalysis(AnalysisType analysisType, Analysis analysis);
 
     public ReferenceSequence importReferenceSequence(String accession) {
+        ReferenceSequence referenceSequence = null;
+        try {
+            String assemblyXml = sraXmlRetrieverByAccession.getXml(accession);
+            AssemblyType assembly = sraAssemblyXmlParser.parseXml(assemblyXml, accession);
+            referenceSequence = referenceSequenceConverter.convert(assembly);
+            Taxonomy taxonomy = taxonomyExtractorFromReferenceSequence.getTaxonomy(assembly.getTAXON());
+            referenceSequence.setTaxonomy(taxonomy);
+            referenceSequence = referenceSequenceRepository.findOrSave(referenceSequence);
+        } catch (Exception exception) {
+            IMPORT_LOGGER.log(Level.SEVERE, "Encountered Exception for ReferenceSequence accession " + accession);
+            IMPORT_LOGGER.log(Level.SEVERE, exception.getMessage());
+        }
+        return referenceSequence;
+    }
+
+    private Set<String> getReferenceSequenceAccessions(AnalysisType analysisType) {
+        Set<String> referenceSequenceAccessions = new HashSet<>();
+        AnalysisType.ANALYSISTYPE analysisType1 = analysisType.getANALYSISTYPE();
+        // ENA records can contain a reference sequence in either of the three analysis categories:
+        // REFERENCE_ALIGNMENT, SEQUENCE_VARIATION, and PROCESSED_READS. All accessions are collected and returned.
+        if (analysisType1.isSetREFERENCEALIGNMENT()) {
+            referenceSequenceAccessions.add(
+                    // TODO: should we use getRefname() or getAccession()?
```

Comment:
All EGA xmls have refName only

Reply:
so I guess we can just use the refName to create the referenceSequence in SraObjectImportThroughDatabase

Reply:
@selvaebi Thank you for the clarification. I changed the implementation so that `getAccession()` is used by default while `getRefname()` is used for import from the database.

@cyenyxe What should we do in the case of ENA? Consider the two following examples:
* Analysis [ERZ496533](https://www.ebi.ac.uk/ena/data/view/ERZ496533&display=xml), which is a `SEQUENCE_VARIATION`, has assembly [GCA_000002305.1](https://www.ebi.ac.uk/ena/data/view/GCA_000002305.1&display=xml) which only has an accession. This is a standard assembly which can be parsed & used as usual.
* Analysis [ERZ004000](https://www.ebi.ac.uk/ena/data/view/ERZ004000&display=xml), which is a `REFERENCE_ALIGNMENT`, has assembly [HAAA01000000](https://www.ebi.ac.uk/ena/data/view/HAAA01000000&display=xml), which only has a refName. Moreover, this is not a standard assembly; it returns an <ENTRY> entity, and I'm not sure what to do with them (and whether we _want_ to do anything with them)

Reply:
We should only process `SEQUENCE_VARIATION` analyses for now, but your question about HAAA01000000 still applies.

That is a transcriptome, one of the 3 types of sequences we support, so it has to be processed. The set of sequences it contains is
> \<xref db="ENA-TSA" id="HAAA01000001-HAAA01003524"/\>

Reply:
@iyangar Transcriptome and gene sequences are still not supported in the latest version of the PR. See example of a transcriptome reference sequence here:
* https://www.ebi.ac.uk/ena/data/view/ERZ004000&display=xml
* https://www.ebi.ac.uk/ena/data/view/HAAA01000000&display=xml

For it, we have an empty `STANDARD` (no accession, only a refname), and the actual accessions are directly inside the `REFERENCE_ALIGNMENT`:

```
<SEQUENCE accession="HAAA01000001.1" label="PNEJI2.1.1"/>
<SEQUENCE accession="HAAA01000002.1" label="PNEJI2.2.1"/>
<SEQUENCE accession="HAAA01000003.1" label="PNEJI2.3.1"/>
...
```

Reply:
correct, In today's standup we decided to discuss about this in sprint meeting